entity/machine.go

```diff
@@ -85,6 +93,53 @@ type Machine struct {
 	Locked                       bool                `json:"locked,omitempty"`
 }
 
+func (m *Machine) UnmarshalJSON(data []byte) error {
+	type machine Machine
+
+	if err := json.Unmarshal(data, (*machine)(m)); err != nil {
+		return err
+	}
+
+	// NOTE: deduce field value for backward compatibility
+	// `enable_hw_sync` field is exposed via API in MAAS 3.4+
+	// but hardware sync feature was added in 3.2
+	if !m.EnableHwSync {
+		t := MAASTime{}
+		if m.HwSyncInterval != 0 && (m.HwNextSync != t || m.HwLastSync != t) {
+			m.EnableHwSync = true
+		}
+	}
+
+	return nil
+}
+
+// MAASTime is a custom time format returned by MAAS API
+// which is not RFC3339
+type MAASTime struct {
```

Comment:
does this need to be a struct or can it be `type MAASTime time.Time`?

Reply:
Should be fixed now. Was experimenting with different options and forgot to rollback. Thanks!